File: Data Structures & Algorithms/binary-tree-right-side-view/submission-3.cpp
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        std::vector<int> res;
        std::queue<TreeNode*> queue;

        if (!root) return res;

        queue.push(root);

        while (!queue.empty()) {
            int elemsOnThisLevel = queue.size();
            for (int i = 0; i < elemsOnThisLevel - 1; i++) {
                auto cur = queue.front();
                queue.pop();

                if (cur->left) queue.push(cur->left);
                if (cur->right) queue.push(cur->right);
            }

            auto cur = queue.front();
            res.push_back(cur->val);
            queue.pop();

            if (cur->left) queue.push(cur->left);
            if (cur->right) queue.push(cur->right);
        }

        return res;
    }
};

File: Data Structures & Algorithms/invert-a-binary-tree/submission-3.py

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root is None:
            return root
        queue = [root]
        while len(queue) > 0:
            cur = queue.pop()
            if cur.left is not None:
                queue.append(cur.left)
            if cur.right is not None:
                queue.append(cur.right)
            
            temp = cur.left
            cur.left = cur.right
            cur.right = temp

        return root